rmw/include/rmw/types.h

```diff
@@ -192,14 +226,30 @@ enum RMW_PUBLIC_TYPE rmw_qos_durability_policy_t
   RMW_QOS_POLICY_DURABILITY_VOLATILE
 };
 
+enum RMW_PUBLIC_TYPE rmw_qos_liveliness_policy_kind_t
+{
+  RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT,
+  RMW_QOS_POLICY_LIVELINESS_AUTOMATIC,
+  // TODO(eknapp) - re-enable these once these types of liveliness are supported
+  // RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_PARTICIPANT,
+  // RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC
+};
+
 /// ROS MiddleWare quality of service profile.
 typedef struct RMW_PUBLIC_TYPE rmw_qos_profile_t
 {
   enum rmw_qos_history_policy_t history;
   /// Size of the message queue.
   size_t depth;
+
   enum rmw_qos_reliability_policy_t reliability;
   enum rmw_qos_durability_policy_t durability;
+  struct rmw_time_t deadline;
+  struct rmw_time_t lifespan;
+
+  enum rmw_qos_liveliness_policy_kind_t liveliness;
```

Comment:
rmw_qos_liveliness_policy_kind_t is a different convention than rmw_qos_durability_policy_t. Should be rmw_qos_liveliness_policy_t. Also has this been defined yet?

Reply:
Durability is only an enum, liveliness is two things, a `kind` and a `lease_duration`, you think we should create a struct instead to carry both items?  The reason that I split it is because `history` QoS was already split between the `enum rmw_qos_history_polity_t` and `size_t depth` members - I was following the standing convention

Reply:
I don't think we need to create a new wrapper stuct. I think Ross' comment was mainly about the naming of the enum type.

Reply:
Okay, let's rename `rmw_qos_liveliness_policy_kind_t` to `rmw_qos_liveliness_policy_t` to be consistent with `rmw_qos_history_policy_t`.